### PR TITLE
updated readme/basic docs

### PR DESCRIPTION
**Pull request by gkiar**
- [x] Update README
- [x] Add contributing guide
- [x] Update PR tempalte
- [x] Update issue template
- [x] Update license

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,49 +1,146 @@
...
+each execution will be evaluated with a unique random state, meaning that when you
+run it again you may get slightly (or very) different results. If your goal is to
+characterize the variability in your results, or obtain a robust estimate of the
+"true" mean answer, you will need to then run your tool multiple times and compare
+each execution.
+
+It's also important to remember that the execution time will be increased when using
+*Fuzzy*, as compared to running tools in a determinitic environment. Depending on
+the tool, the instrumentation, and the MCA mode, this additional overhead may range
+from negligible to the order of a ***100x*** slowdown.
```
Comment (yohanchatelain) [inline README.md:92]: Now it's more a x30 slowdown

File: README.md [hunk 2]
```diff
@@ -1,49 +1,146 @@
...
+```
+docker run -ti verificarlo/fuzzy
+```
+
+If you're on a shared system and would prefer to use [Singularity](https://sylabs.io/),
+that's no problem, just convert the container using the appropriate method for
+your system ([e.g.](https://docs.computecanada.ca/wiki/Singularity#Creating_images)).
+
+If you would like to build the environment locally on your system, look at the
+Dockerfiles in `docker/base/` to see how installation was performed.
```
Comment (glatard) [inline README.md:51]: Maybe add a few words explaining what this image provides?

File: README.md [hunk 3]
```diff
@@ -1,49 +1,146 @@
...
+
+#### Adding your software
+If your code is written in, say, Python, then you can simply mount your code to
+this environment, run it using the contained interpreter, and experince the *:fuzz:*.
+If your code loads shared libraries, such as `libmath` or `lapack`, make sure that
```
Comment (glatard) [inline README.md:56]: We should refer to libmath as "GNU C Library" and link to https://www.gnu.org/software/libc/manual/html_node/Mathematics.html

File: README.md [hunk 4]
```diff
@@ -1,49 +1,146 @@
...
+docker run -ti verificarlo/fuzzy
+```
+
+If you're on a shared system and would prefer to use [Singularity](https://sylabs.io/),
+that's no problem, just convert the container using the appropriate method for
+your system ([e.g.](https://docs.computecanada.ca/wiki/Singularity#Creating_images)).
+
+If you would like to build the environment locally on your system, look at the
+Dockerfiles in `docker/base/` to see how installation was performed.
+
```
Comment (yohanchatelain) [inline README.md:52]: Could add a basic example, like:
```bash
$ python3 -c "print([sum([.001]*1000) for _ in range(3)])"
[1.0, 0.9999999999999997, 1.0000000000000007, 1.0000000000000002]
```

**Review by yohanchatelain — APPROVED** [review]
LGTM!

File: README.md (hunk 3 above)
Comment (yohanchatelain) [reply README.md:56]: Why refering to GNU C lib? People can use whatever libmath they want

Comment (glatard) [reply README.md:56]: fair point

File: README.md [hunk 5]
```diff
@@ -1,49 +1,146 @@
...
+In the context of *fuzzy* experiments, it is important to remember that by default
+each execution will be evaluated with a unique random state, meaning that when you
+run it again you may get slightly (or very) different results. If your goal is to
+characterize the variability in your results, or obtain a robust estimate of the
+"true" mean answer, you will need to then run your tool multiple times and compare
+each execution.
+
+It's also important to remember that the execution time will be increased when using
+*Fuzzy*, as compared to running tools in a determinitic environment. Depending on
+the tool, the instrumentation, and the MCA mode, this additional overhead may range
```
Comment (shnizzedy) [inline README.md:91]: MCA is "Monte-Carlo Arithmetic"? PB is "Protocol Buffers"? I might be missing it somewhere, but I don't see what the options for the perturbation modes are

Comment (yohanchatelain) [reply README.md:91]: Thanks for pointing out. Options are available at the [verificarlo](https://github.com/verificarlo/verificarlo/blob/master/README.md#mca-backend-libinterflop_mcaso) project. We should add a section for them here.

Comment (yohanchatelain) [reply README.md:91]: To answer MCA stands for Monte Carlo Arithmetic. In this stochastic arithmetic you have three modes for perturbing your floating-point operations `x = a op b` where op in {+,-,*,/}: 
- Random Rounding (RR): `x = inexact(a op b)`
- Precision Bounding (PB): `x = inexact(a) op inexact(b)`
- Full MCA (MCA): `x = inexact(inexact(a) op inexact(b))`

File: README.md [hunk 6]
```diff
@@ -1,49 +1,169 @@
...
+
+A simple sanity check to verify that your code is using the perturbed libraries
+is to dramatically reduce the precision in your configuration by changing the
+`precision-binary32` and `precision-binary64` values to something small (e.g. 1),
+boot up a debugging session in your environment (e.g. a Python shell, GDB, etc.),
+load a math library, and run a basic math function a few times (e.g. `sin(1)`) —
+you should get different results if your instrumentation is setup properly.
+***Important***: Don't forget to set the precision back to your desired level
+prior to performing your experiments!
+
```
Comment (SucyLi) [inline README.md:88]: How to make sure the config actually changes the precision after we generate the config? A concrete step-by-step example would be appreciated

Comment (gkiar) [reply README.md:90]: Thanks!

**Pull request merged**